types/rest/rest_test.go

```diff
@@ -55,3 +55,72 @@ func TestBaseReqValidateBasic(t *testing.T) {
 		})
 	}
 }
+
+func TestParseHTTPArgs(t *testing.T) {
+	req0, err := http.NewRequest("", "/", nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	req1, err := http.NewRequest("", "/?limit=5", nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	req2, err := http.NewRequest("", "/?page=5", nil)
```

Comment:
Let's remove some redundant error-checking code, shall we? First, drop this function somewhere in this file (usually: at the very bottom):

```golang
func mustNewRequest(t *testing.T, method, url string, body io.Reader) *http.Request {
    req, err := http.NewRequest(method, url, body)
    require.NoError(t, err)
    return req
}
```

Then replace all calls like:

```golang
	req1, err := http.NewRequest("", "/?limit=5", nil)
	if err != nil {
		t.Fatal(err)
	}
```

with:

```golang
    req1 := mustNewRequest(t, "", "/?limit=5", nil)
```

Reply:
Thanks man! quality bump 🆙